### ISSUE TITLE
Using IDs instead of names to identify resources

### ISSUE THREAD

**Issue by pablo-de-andres** [materials-marketplace/standard-app-api#45]
Names are used in the path parameters for the operations with datasets and collections (`dataset_name`, `collection_name`). However, these might not be unique.

Raised by @Kirankumaraswamy.

**Comment by pablo-de-andres** [comment]
@csadorf thoughts?

**Comment by csadorf** [comment]
Could you provide an example of the issue? It is not immediately clear to me what the problem is.

**Comment by pablo-de-andres** [comment]
For instance [here](https://github.com/materials-marketplace/standard-app-api/blob/main/marketplace_standard_app_api/routers/object_storage.py#L273), to get a dataset the parameters are `collection_name` and `dataset_name`.

But I actually see in the [datamodel](https://github.com/materials-marketplace/standard-app-api/blob/main/marketplace_standard_app_api/models/object_storage.py#L31) that this is actually an id.

The confusion came from the usage of "name", that does not necessarily transmit the need for them to be unique.

I would say the point is then whether it shoulb renamed or left as is.

**Comment by csadorf** [comment]
Why don't we just add a note to the description clarifying that the collection name and data set name are unique? There is no way to accidentally use the API in a wrong way here, but maybe it helps to clear up any confusion?

**Comment by Kirankumaraswamy** [comment]
Here the collection names and dataset names are folder names and file names in a directory. Hence they don't need to be unique.

**Comment by csadorf** [comment]
> Here the collection names and dataset names are folder names and file names in a directory. Hence they don't need to be unique. We can have a file with the same name in some other directory.

Yes, but that would be ok? Different collections can have data sets with the same name. They just have to be unique within the same collection.

**Comment by Kirankumaraswamy** [comment]
You are right but when we retrieve a single dataset ,we will retrieve it by using either dataest_id or dataset_title. We are not using collection_id to retrieve it. When we request for a single dataset with a title then we have to send all the dataset with matching titles across all collections.

**Comment by pablo-de-andres** [comment]
@csadorf is not working on this anymore.

Who is _we_ in "when we retrieve"? To me it makes sense that the collection is needed, similar to how the path/folder is needed to access a file.

**Comment by Kirankumaraswamy** [comment]
We in the sense DataSink. Currently python-sdk doesn't support using both collection ID and dataset ID to retrieve the dataset. Same with Datasink side. We need to change the python-sdk code again.

**Comment by pablo-de-andres** [comment]
The [python SDK](https://github.com/materials-marketplace/python-sdk/blob/main/marketplace/app/v0/object_storage.py#L121) does support it, does it not?
